packages/core/src/defer/triggering.ts

```diff
@@ -433,6 +459,32 @@ export async function triggerHydrationFromBlockName(
   );
 }
 
+export function deferBlockHasErrored(lView: LView, tNode: TNode): boolean {
+  return getLDeferBlockDetails(lView, tNode)[DEFER_BLOCK_STATE] === DeferBlockState.Error;
+}
+
+function cleanupParentContainer(
+  i: number,
+  hydrationQueue: string[],
+  dehydratedBlockRegistry: DehydratedBlockRegistry,
+) {
+  if (i > 0 && dehydratedBlockRegistry.get(hydrationQueue[i - 1]) != null) {
+    const parentDeferBlock = dehydratedBlockRegistry.get(hydrationQueue[i - 1])!;
+    cleanupLContainer(parentDeferBlock.lContainer);
+  }
```

Comment:
nit: minor refactoring proposal:
```suggestion
  // If a parent block exists, it's in the hydration queue in front of the current block.
  const parentDeferBlockIdx = currentBlockIdx - 1;
  const parentDeferBlock = parentBlockIdx >= 0 ? dehydratedBlockRegistry.get(hydrationQueue[parentBlockIdx]) : null;
  if (parentDeferBlock) {
    cleanupLContainer(parentDeferBlock.lContainer);
  }
```